src/solvers/nonlinear_solid_operators.hpp

```diff
@@ -114,56 +158,53 @@ class NonlinearSolidReducedSystemOperator : public mfem::Operator {
    * @brief Essential degrees of freedom
    */
   const std::vector<serac::BoundaryCondition>& ess_bdr_;
+};
 
+/**
+ * @brief The abstract time dependent MFEM operator for explicit and implicit solves
+ */
+class NonlinearSolidDynamicOperator : public mfem::TimeDependentOperator {
 public:
   /**
-   * @brief Construct a new Nonlinear Solid Reduced System Operator object
+   * @brief Construct a new Nonlinear Solid Dynamic Operator object
    *
    * @param[in] H_form The nonlinear stiffness form
    * @param[in] S_form The linear viscosity form
    * @param[in] M_form The linear mass form
    * @param[in] ess_bdr The essential boundary conditions
+   * @param[in] newton_solver The newton solver object
+   * @param[in] lin_params The linear solver parameters
    */
-  NonlinearSolidReducedSystemOperator(const mfem::ParNonlinearForm& H_form, const mfem::ParBilinearForm& S_form,
-                                      mfem::ParBilinearForm&                       M_form,
-                                      const std::vector<serac::BoundaryCondition>& ess_bdr);
+  NonlinearSolidDynamicOperator(std::unique_ptr<mfem::ParNonlinearForm>      H_form,
+                                std::unique_ptr<mfem::ParBilinearForm>       S_form,
+                                std::unique_ptr<mfem::ParBilinearForm>       M_form,
+                                const std::vector<serac::BoundaryCondition>& ess_bdr, mfem::NewtonSolver& newton_solver,
+                                const serac::LinearSolverParameters& lin_params);
 
   /**
-   * @brief Set current dt, v, x values - needed to compute action and Jacobian.
+   * @brief Evaluate the explicit time derivative
    *
-   * @param[in] dt The current timestep
-   * @param[in] v The current velocity
-   * @param[in] x The current position
+   * @param[in] vx The current velocity and displacement state
+   * @param[out] dvx_dt The explicit time derivative of the state vector
    */
-  void SetParameters(double dt, const mfem::Vector* v, const mfem::Vector* x);
+  virtual void Mult(const mfem::Vector& vx, mfem::Vector& dvx_dt) const;
 
   /**
-   * @brief Compute y = H(x + dt (v + dt k)) + M k + S (v + dt k).
+   * @brief Solve the Backward-Euler equation: k = f(x + dt*k, t), for the unknown k.
```

Comment:
Information is being lost by changing these comments. This object isn't computing some arbitrary f(x,t), it's doing something specific, namely `//Compute y = H(x + dt (v + dt k)) + M k + S (v + dt k).`